java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/SparkPlanner.scala

```diff
@@ -338,12 +339,39 @@ class SparkPlanner(session: SparkSession, config: OpenmldbBatchConfig, sparkAppN
     }
 
     try {
-      engine = new SqlEngine(sql, dbs, engineOptions, config.defaultDb)
-      val res = body(engine)
+      sqlEngine = new SqlEngine(dbs, engineOptions)
+      val engine = sqlEngine.getEngine
+
+      // TODO(tobe): If use SparkPlanner instead of OpenmldbSession, these will be null
+      if (config.openmldbZkCluster.nonEmpty && config.openmldbZkRootPath.nonEmpty
+        && openmldbSession != null && openmldbSession.openmldbCatalogService != null) {
+        val externalFunMap = openmldbSession.openmldbCatalogService.getExternalFunctionsMap()
+        for ((functionName, functionProto) <- externalFunMap){
+          logger.info("Register the external function: " + functionProto)
+          val returnDataType = DataTypeUtil.protoTypeToOpenmldbType(functionProto.getReturnType)
+          val argsDataType = new VectorDataType()
+          functionProto.getArgTypeList.forEach(dataType => {
+            argsDataType.add(DataTypeUtil.protoTypeToOpenmldbType(dataType))
+          })
+
+          // Get the correct file path
+          val soFilePath = functionProto.getName.split("/").last
+          engine.RegisterExternalFunction(functionName, returnDataType, argsDataType, false, soFilePath)
```

Comment:
get `is_aggregate` from `functionProto` and pass to `RegisterExternalFunction` in order to support udaf automatically in the future

Reply:
Okay. Resolved.